nntrainer/layers/layer_node.h

```diff
@@ -307,6 +307,30 @@ class LayerNode : public ml::train::Layer, public GraphNode {
    */
   friend std::ostream &operator<<(std::ostream &out, const LayerNode &l);
 
+  /**
+   * @brief   Get init layer context
+   *
+   * @retval  init layer context
+   */
+  const InitLayerContext &getInitContext() const { return init_context; }
+
+  /**
+   * @brief   Get run layer context
+   *
+   * @retval  run layer context
+   */
+  const RunLayerContext &getRunContext() const { return run_context; }
+
+  /**
+   * @brief   Set run layer context
+   *
+   * @param  context Updated run layer context
+   */
+  void updateRunContext(const RunLayerContext &context) {
+    // TODO: ensure props/trainable must match
+    run_context = std::move(context);
```

Comment:
Since move is a casting operation, we need signature of `RunLayerContext &&context` or `RunLayerContext &context` (non-const)
Otherwise copy will be invoked.

Reply:
> Otherwise copy will be invoked.

can copy be invoked? as there is unique_ptr inside and that will give compile error.

Reply:
https://wandbox.org/permlink/JtK4a4npPR3bXtYc check this out.

It seems like unique_ptr does not exist inside context and nor move ctor/assign operator as well.

Reply:
quite strange. Updated to `RunLayerContext &&context`.